### DISCUSSION TITLE
Free format Fortran UMAT

### DISCUSSION THREAD

**Discussion by Flolaffel** [idaholab/moose#21476] [Q&A Modules: Solid mechanics] [answered]
Hello,

when running `make` to compile my UMAT I get

`Can't open included file 'xxx.f90'`

Does that mean that MOOSE does not support free format UMATs?

**Comment by GiudGiud** [comment]
not sure if it does
@dschwen @recuero @ngrilli will know. 
If it does and the problem is local we ll want to see the output of `make -n`

**Reply by recuero** [reply]
@Flolaffel have you tried passing the compiler flag `-ffree-form`?

**Reply by Flolaffel** [reply]
I'm not sure, what you mean by using `-ffree-form`. To me it seems like that's not an available option when using the `make` command. What are you refering to?

**Reply by GiudGiud** [reply]
You can add additional compiling and linking flags by modifying the environment variables. 
You may also modify the Makefile used to compile the UMAT

**Reply by jessecarterMOOSE** [reply]
FWIW it looks like there's a Makefile variable `PLUGIN_FLAGS` that you can set to influence how your UMAT gets compiled. I set this in my app's Makefile to change the line length etc. You can see what else goes into building UMAT plugins [here](https://github.com/idaholab/moose/blob/next/framework/build.mk#L306-L322).

**Comment by Flolaffel** [comment, marked as answer]
Update: I have to admit I don't have enough knowledge about UMATs, Fortran and compiler flags to solve this problem on my own. The UMAT I'm working with is written (not by me) in a mix of F77 and F90. From @jessecarterMOOSE's comment I saw that MOOSE provides options to compile F90. However in my case `make` always exited with an error when it came to compile the F90 part of the UMAT. So I went and asked my master thesis supervisor for help.

Here is how he solved it: He used @GiudGiud's recommended `make -n ` command to get get the flags MOOSE uses for compiling. Then, following those commands, he wrote a script that creates the .plugin file you need for MOOSE externally. You can then simply copy that file into the MOOSE plugins folder and it seems to work just fine.
